### ISSUE TITLE
Enhancement requst: axes_manager should have a convenience method for easy calibration

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#2237]
A colleague of mine came across a confusing issue when trying to change the units of a signal that did not have the axes already calibrated. 

Given the axes manager:

Navigation axis name | size | index | offset | scale | units
-- | -- | -- | -- | -- | --
phase | 10 | 0 | 0.0 | 20.0 | degrees

Signal axis name | size | offset | scale | units
-- | -- | -- | -- | --
x | 3956 | 0.0 | 1.0 |  
y | 2574 | 98.0 | 1.0

Running `s.axes_manager.convert_units(axes="x", units="um", same_units=True, factor=19.512195)` results in a `DimensionalityError: Cannot convert from 'dimensionless' (dimensionless) to 'micrometer' ([length])`.

While this error is descriptive, it would be nice if the code could "force" the units as given if there is no calibration currently defined. It would allow for slightly more concise code than:

```python
s.axes_manager[0].name = 'x'
s.axes_manager[0].scale = 19.512195
s.axes_manager[0].units = 'um'
```

What are others' thoughts?

**Comment by ericpre** [comment]
I would say that not allowing to use `convert_units` for an axis which is not already calibrated is a good behaviour and the error message is fairly clear.
What would possibly be more suitable is a `calibrate` method and there is already [one](https://github.com/hyperspy/hyperspy/blob/d856f2af233d5d4d5aac096d51b834666135273b/hyperspy/axes.py#L553) for the `DataAxis` but it is not documented and it does not seem to do what you want here. So maybe it would worth thinking about extending this method and add a similar one to the `AxesManager`.

**Comment by jat255** [comment]
I agree; I think having an easily used `calibrate` method would be a good idea, and would be more concise. Then if `convert_units` fails, it could point the user to `calibrate` first.

**Comment by francisco-dlp** [comment]
@jat255, would you like to close this then?

**Comment by jat255** [comment]
@francisco-dlp alright to leave it on the wish list? I think there's an opportunity for enhancement here, I just haven't gotten around to it yet.

**Comment by francisco-dlp** [comment]
Sounds good, but in that case it would be good to change the title to something that describes better the improvement that you are proposing.
